### PR TITLE
chore: make fedimintd and devimint shutdown stuff fast

### PR DESCRIPTION
**Pull request by dpc**
See commits.

### Testing

Start devimint with:

```
env RUST_LOG=fm=debug just devimint-env
```

`exit` in it, and observe the shutdown time. `tail -f target-nix/devimint/logs/fedimintd-0.log` if you want to see what's happening in `fedimintd`. Another way is to `killall fedimintd` from the outside with `tail -f`.

### CODE REVIEW

File: fedimintd/src/fedimintd.rs [hunk 1]
```diff
@@ -331,30 +330,24 @@ impl Fedimintd {
 
         let timing_total_runtime = timing::TimeReporter::new("total-runtime").info();
 
-        // DO NOT REMOVE, or spawn_local tasks won't run anymore
-        let local_task_set = tokio::task::LocalSet::new();
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:335]: I guess this is not needed anymore, probably after alephbft refactoring.

File: devimint/src/util.rs [hunk 2]
```diff
@@ -107,31 +100,60 @@ pub struct ProcessHandleInner {
...
+                target: LOG_DEVIMINT,
+                name=%self.name,
+                signal="SIGTERM",
+                "sending signal to terminate child process"
+            );
+
+            send_sigterm(child);
+
+            if (fedimint_core::runtime::timeout(Duration::from_secs(2), child.wait()).await)
+                .is_err()
```
Comment (maan2003) [inline devimint/src/util.rs:116]: given fedimintd sleeps for 10s on shutdown, this will always send SIGKILL?

Comment (dpc) [reply devimint/src/util.rs:116]: I fixed that. Now it shutdowns in <1s.

File: fedimintd/src/fedimintd.rs [hunk 3]
```diff
@@ -367,14 +360,8 @@ impl Fedimintd {
                     sleep(SHUTDOWN_TIMEOUT).await;
```
Comment (maan2003) [inline fedimintd/src/fedimintd.rs:360]: aren't we always sleeping for 10s herre

Comment (dpc) [reply fedimintd/src/fedimintd.rs]: I don't understand what happened. I'm quite sure I was not seeing it.

But anyway - I got rid of this now, and also nested api runtime shenanigans that were causing a panic.

**Review by maan2003 — APPROVED** [review]

**Comment by dpc** [comment]
Seems like this saves 2 mins on test-ci-all in the CI, from around 13:20 to 11:20.

**Comment by maan2003** [comment]
needs further review for 1419130

**Pull request merged**